src/class-block-converter.php

```diff
@@ -315,21 +318,30 @@ public function remove_image_args( $url ): string {
 		// Split url.
 		$url_parts = wp_parse_url( $url );
 
-		return $url_parts['scheme'] . '://' . $url_parts['host'] . $url_parts['path'];
+		if ( isset( $url_parts['query'] ) ) {
+			$url = str_replace( '?' . $url_parts['query'], '', $url );
+		}
+		if ( isset( $url_parts['fragment'] ) ) {
+			$url = str_replace( '#' . $url_parts['fragment'], '', $url );
+		}
+		if ( str_starts_with( $url, '//' ) ) {
+			$url = preg_replace( '#^//#', 'https://', $url );
```

Comment:
Would be better to get the currently configured `home` value from options and use the scheme there. I know it's 2023 and everyone _should_ be using https, but for safety, let's mirror the protocol used by the site as configured in options.

Reply:
I updated this function to include this, and to invert the logic to put the pieces back together using only the values we care about.